url2df0f7c811_przemekgalazka_legacycode_tgz-pJ8-pl_cc_core_cmd_CommandJ8/src/ccproxy/MessageFromAsterisk.java

```diff
@@ -0,0 +1,492 @@
+package ccproxy;
+
+import pl.cc.core.cmd.Command;
+import pl.cc.core.cmd.EventProxyClientPause;
+import pl.cc.exceptions.AgentNotFoundException;
+import pl.cc.exceptions.QueueNotFoundException;
+import pl.cc.real.RealAgent;
+import pl.cc.real.RealCall;
+import pl.cc.real.RealQueue;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Iterator;
+
+import org.apache.log4j.Logger;
+
+public class MessageFromAsterisk {
+  org.apache.log4j.Logger log = Logger.getLogger(MessageFromAsterisk.class);
+  String eventMsg;
+  String exten;
+
+  String event = null;
+  String response = null;
+  Events e;
+  boolean showQueueMember = true;
+  HashMap<String, String> messageHash = new HashMap<String, String>();
+
+  Parametry p;
+
+  AgentResponseProcessor agentResponsProcessor;
+
+
+  public MessageFromAsterisk(Parametry params) {
+    this(params, new Events(params));
+  }
+
+
+  public MessageFromAsterisk(Parametry p, Events e) {
+    this.p = p;
+    this.e = e;
+    this.agentResponsProcessor = new AgentResponseProcessor(p, e);
+  }
+
+
+  /**
+   * @param line Linia otrzymana od Asteriska, ktora nalezy sparsowac.
+   */
+  public void parseLine(String line) {
+    if (line.indexOf(": ") == -1) {
+      log.info("Linia od asteriska bez dwukropka: " + line);
+      return;
+    }
+    String key = line.substring(0, line.indexOf(": "));
+    String param = line.substring(line.indexOf(": ") + 1);
+    param = param.trim();
+    if (key.equals("Event"))
+      event = param;
+    if (param != null)
+      messageHash.put(key.toLowerCase(), param);
+    else
+      messageHash.put(key, "");
+  }
+
+
+  public void showEvent() {
+    if (p.c.logAsteriskEvents) {
+      Iterator<String> i = messageHash.values().iterator();
+      Iterator<String> j = messageHash.keySet().iterator();
+
+      p.log.info("--------Event begin");
+      p.log.info(event);
+      while (i.hasNext()) {
+        p.log.info(j.next() + " -> " + i.next());
+      }
+      p.log.info("--------Event end");
+    }
+  }
+
+
+  private String getHeader(String key) {
+    String value = messageHash.get(key.toLowerCase());
+    return value;
+  }
+
+
+  public void parseEvent(WiadomoscOdAsteriskaList wiadomosciOdAsteriska) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 210 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 197-437 into a separate method.